content/src/debug.js

```diff
@@ -123,15 +134,17 @@ var debug = window.ide = {
       panel: embedded ? 'auto' : true
     };
   },
-  trace: function(event, data) {
+  trace: function(event, data) { 
     detectStuckProgram();
     // This receives events for the new debugger to use.
     currentDebugId += 1;
-    var record = {line: 0, eventIndex: null, startCoords: [], endCoords: [], method: "", data: "", seeeval:false};
+    var record = {line: 0, eventIndex: null, startCoords: [], endCoords: [], method: "", 
+        data: "", seeeval:false};
     traceEvents.push(event);
     currentEventIndex = traceEvents.length - 1;
     record.eventIndex = currentEventIndex;
     var lineno = traceEvents[currentEventIndex].location.first_line;
+    setTimeout(function() {view.createSlider(traceEvents, arrows, view.paneid("left"), debugRecordsByLineNo, targetWindow)}, 1000);
```

Comment:
Done, setupSlider() is currently being used in debug.js and now all events relating to the slider are handled in debug.js. Only an array of line numbers are given to createSlider in view.js for visual elements.